### ISSUE TITLE
place-holder for some observed oddities and opportunities

### ISSUE THREAD

**Issue by matsc-at-sics-se** [matsc-at-sics-se/unison#34]
`sp_process_finalizer` was oddly non-optimal

`compressBound` - presolver crash

`SP_install_fd_hooks` illustrates a nice transformation

Adjust `RSP` by 8 should be dominated by spill callee-saved

`MOV32ri 0` --> `XOR32rr` if eflags are dead

**Comment by matsc-at-sics-se** [comment]
`clear_ch_type_hash`: `ADD32ri_LEA` was not selected

**Comment by matsc-at-sics-se** [comment]
`spio_coder_default_release`: oddly non-optimal

**Comment by matsc-at-sics-se** [comment]
`putShortMSB`: oddly non-optimal

**Comment by matsc-at-sics-se** [comment]
`atom_buffer_init`: seems to spill 4 bytes, then unspill 8 bytes, expecting zero-extension

**Comment by matsc-at-sics-se** [comment]
`checkdealloc_internal`: presolver crash

**Comment by matsc-at-sics-se** [comment]
`utf32_decoder_seek`: suboptimal

**Comment by matsc-at-sics-se** [comment]
`label_insn_i`: suboptimal

**Comment by matsc-at-sics-se** [comment]
`term_heap_offset`: suboptimal

**Comment by robcasloz** [comment]
Opportunity: implement swaps (`xchg`) as parallel register-to-register moves. See this case study:
```
extern void bar(int y, int x);

void foo(int x, int y) {
  bar(y, x);
}
```
LLVM generates a three-way swap:
```
	movl	%edi, %eax
	movl	%esi, %edi
	movl	%eax, %esi
	jmp	bar                     # TAILCALL
```
where the following code would be better:
```
	xchgl	%edi, %esi
	jmp	bar                     # TAILCALL
```

**Comment by matsc-at-sics-se** [comment]
Presolver sub-issues fixed by commits `c14498887f6bfc9d10afafcecbef2bb9131bd6f7` and `eeef0945289b3a42d01b09bfa012ad35a2312fd7`.

**Comment by matsc-at-sics-se** [comment]
More sub-issues fixed by commits `ed7fdaa97cf746ab1eee2c3d7e05bfdf7ef87a61`, `890cdd128992c4ba0347823d7f6df7f6127428e1`, `0f8fc4a8d85a99b928b403e01fb48448f7597c6e`, `9622d7b01a9e0a5cae1ca78d5024fda6502d60f3`, `047cae5076730074f15d9fa7c6ffc66a7c561f96`.
